recipes/aravis/all/conanfile.py

```diff
@@ -0,0 +1,170 @@
+        if not self.options["glib"].shared and self.options.shared:
+            raise ConanInvalidConfiguration("Shared Aravis cannot link to static GLib")
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("macOS builds are disabled until conan-io/conan#7324 gets merged to fix macOS SIP issue #8443")
+
+    def build_requirements(self):
+        self.build_requires("meson/0.60.2")
+        self.build_requires("pkgconf/1.7.4")
+        if self.options.introspection:
+            self.build_requires("gobject-introspection/1.70.0")
+
+    def requirements(self):
+        self.requires("glib/2.70.1")
+        self.requires("libxml2/2.9.12")
+        self.requires("zlib/1.2.11")
+        if self.options.usb:
+            self.requires("libusb/1.0.24")
+        if self.options.gst_plugin:
+            self.requires("gstreamer/1.19.2")
+            self.requires("gst-plugins-base/1.19.2")
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_meson(self):
+        if self._meson:
+            return self._meson
+        defs = dict()
+        defs["wrap_mode"] = "nofallback"
+        defs["usb"] = "enabled" if self.options.usb else "disabled"
+        defs["gst-plugin"] = "enabled" if self.options.gst_plugin else "disabled"
+        defs["packet-socket"] = "enabled" if self.options.get_safe("packet_socket") else "disabled"
+        defs["introspection"] = "enabled" if self.options.introspection else "disabled"
+        defs["viewer"] = "disabled"
+        defs["tests"] = "false"
+        defs["documentation"] = "disabled"
+        if self.settings.get_safe("compiler.runtime"):
+            defs["b_vscrt"] = str(self.settings.compiler.runtime).lower()
+        self._meson = Meson(self)
+        self._meson.configure(defs=defs, source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+        return self._meson
+
+    def build(self):
+        self._patch_sources()
+        with tools.environment_append(RunEnvironment(self).vars):
+            meson = self._configure_meson()
+            meson.build()
+
+    def _fix_library_names(self, path):
+        # https://github.com/mesonbuild/meson/issues/1412
+        if not self.options.shared and self._is_msvc:
+            with tools.chdir(path):
+                for filename_old in glob.glob("*.a"):
+                    filename_new = filename_old[3:-2] + ".lib"
+                    self.output.info("rename %s into %s" % (filename_old, filename_new))
+                    shutil.move(filename_old, filename_new)
```

Comment:
```suggestion
                    tools.rename(filename_old, filename_new)
```

Because on Windows it may fail sometimes: https://docs.conan.io/en/latest/reference/tools.html

Reply:
Thanks for reviewing. Didn't know about that one, fixed!